specifications/sep-payment-channel-mechanism.md

```diff
@@ -0,0 +1,631 @@
+  - One `CREATE_CLAIMABLE_BALANCE` operation for each trustline that is
+  disbursing funds to R.
+  - One `END_SPONSORING_FUTURE_RESERVES` operation that confirms reserve
+  account V's sponsorship.
+  - One or more `SET_OPTIONS` operation adjusting escrow account E's
+  thresholds to give I full control of E, and removing R's signers.
+  - One or more `SET_OPTIONS` operation adjusting reserve account E's
+  thresholds to give R full control of V, and removing I's signers.
+
+- D_i, the _declaration transaction_, declares an intent to execute the
+corresponding closing transaction C_i.  D_i has source account E, sequence
+number s_i, and `minSeqNum` set to s_e.  Hence, D_i can execute at any time, so
+long as E's sequence number n satisfies s_e <= n < s_i.  Because C_i has source
+account E and sequence number s_i+1, D_i leaves E in a state where C_i can
+execute.
+
+  D_i does not require any operations, but since Stellar disallows empty
+  transactions, it contains a `BUMP_SEQUENCE` operation with sequence value 0
+  as a no-op.
+
+#### Coordinated Close
+
+Participants can agree to close the channel immediately by modifying and
+resigning the most recently signed confirmation transaction. The participants
+change `minSeqAge` and `minSeqLedgerGap` to zero.
+
+1. Submit most recent D_i
+2. Modify the most recent C_i `minSeqAge` and `minSeqLedgerGap` to zero
+3. Resign and exchange the modified confirmation transaction C_i
+4. Submit modified C_i
+
+#### Uncoordinated Close
+
+Participants can close the channel after the observation period O without
+coordinating. They do this by submitting the most recently signed declaration
+transaction, waiting the observation period O, then submitting the closing
+transaction.
+
+1. Submit most recent D_i
+2. Wait observation period O
+3. Submit C_i
+
+#### Contesting a Close
+
+Participants can contest a close if the close is not the most recent agreed
+closing state of the payment channel.
+
+Participants can attempt to close the channel at a state that is earlier in the
+history of the channel than the most recently agreed to state. A participant who
+is a malicious actor might attempt to do this if an earlier state benefits them.
+
+The malicious participant can do this by performing the [Uncoordinated
+Close](#Uncoordinated-Close) process with a declaration transaction that is not
+the most recently signed declaration transaction.
+
+The other participant can identify that the close process has started at an
+earlier state by monitoring changes in escrow account E's sequence. If the other
+participant sees the sequence number of escrow account E change to a value that
+is not the most recently used s_i, they can use the following process to contest
+the close. A participant contests a close by submitting a more recent
+declaration transaction and closing the channel at the actual final state.
+
+1. Get E's sequence number n
+2. If s_{e+1} >= n < s_i, go to step 3, else go to step 1
```

Comment:
`s_{e+m}`

Reply:
If we wanted to use m here it would be s_e+m, but the intention is to communicate the seq for the next iteration after e, which I think is clearer as it is. If m ever changed too it means the boosting here doesn't have to deal with that.